src/role_nextroomer.js

```diff
@@ -217,25 +217,7 @@ roles.nextroomer.settle = function(creep) {
   }
 
   if (_.sum(creep.carry) === 0) {
-    let hostileStructures = creep.room.findPropertyFilter(FIND_HOSTILE_STRUCTURES, 'structureType', [STRUCTURE_RAMPART, STRUCTURE_EXTRACTOR, STRUCTURE_WALL, STRUCTURE_CONTROLLER], true);
-    if (hostileStructures.length) {
-      let structure = _.max(hostileStructures, s => s.structureType === STRUCTURE_STORAGE);
-
-      if (structure.structureType === STRUCTURE_STORAGE) {
-        if (structure.store.energy === 0) {
-          structure.destroy();
-          return true;
-        }
-      } else if (!structure.energy) {
-        structure.destroy();
-        return true;
-      }
-      creep.say('ho: ' + structure.pos, true);
-      creep.log(structure.structureType);
-      creep.moveTo(structure);
-      creep.withdraw(structure, RESOURCE_ENERGY);
-      return true;
-    }
+    return creep.getEnergyFromHostileStructures();
```

Comment:
I think this is anyway included in `Creep.getEnergy`, so I'm not sure if it is needed here.

And if, I guess if should be 
```
if (creep.getEnergyFromHostileStructures()) {
  return true;
}
```
So that the rest is executed if `getEnergyFromHostileStructures` fails.